packages/ui/src/helpers/authenticator/defaultAuthHubHandler.ts

```diff
@@ -28,14 +29,19 @@ export const defaultAuthHubHandler: AuthMachineHubHandler = (
       send('SIGN_IN_WITH_REDIRECT');
       break;
     }
-    case 'signedOut':
-    case 'tokenRefresh_failure': {
-      if (event === 'signedOut' && isFunction(onSignOut)) {
+    case 'signedOut': {
+      if (isFunction(onSignOut)) {
         onSignOut();
       }
       send('SIGN_OUT');
       break;
     }
+    case 'tokenRefresh_failure': {
+      if (data?.error?.name !== AmplifyErrorCode.NetworkError) {
+        send('SIGN_OUT');
+      }
+      break;
```

Comment:
Nit: Prefer explicit handling on truthiness for network error and default to signing out:
```suggestion
      if (data?.error?.name === AmplifyErrorCode.NetworkError) {
        return;
      }
      send('SIGN_OUT');
      break;
```